src/Generated/RegisterLookup.inc

```diff
@@ -0,0 +1,269 @@
+    /* TR4        */ { ZYDIS_REGCLASS_TEST, 4, 32, 32 },
+    /* TR5        */ { ZYDIS_REGCLASS_TEST, 5, 32, 32 },
+    /* TR6        */ { ZYDIS_REGCLASS_TEST, 6, 32, 32 },
+    /* TR7        */ { ZYDIS_REGCLASS_TEST, 7, 32, 32 },
+    /* CR0        */ { ZYDIS_REGCLASS_CONTROL, 0, 32, 64 },
+    /* CR1        */ { ZYDIS_REGCLASS_CONTROL, 1, 32, 64 },
+    /* CR2        */ { ZYDIS_REGCLASS_CONTROL, 2, 32, 64 },
+    /* CR3        */ { ZYDIS_REGCLASS_CONTROL, 3, 32, 64 },
+    /* CR4        */ { ZYDIS_REGCLASS_CONTROL, 4, 32, 64 },
+    /* CR5        */ { ZYDIS_REGCLASS_CONTROL, 5, 32, 64 },
+    /* CR6        */ { ZYDIS_REGCLASS_CONTROL, 6, 32, 64 },
+    /* CR7        */ { ZYDIS_REGCLASS_CONTROL, 7, 32, 64 },
+    /* CR8        */ { ZYDIS_REGCLASS_CONTROL, 8, 32, 64 },
+    /* CR9        */ { ZYDIS_REGCLASS_CONTROL, 9, 32, 64 },
+    /* CR10       */ { ZYDIS_REGCLASS_CONTROL, 10, 32, 64 },
+    /* CR11       */ { ZYDIS_REGCLASS_CONTROL, 11, 32, 64 },
+    /* CR12       */ { ZYDIS_REGCLASS_CONTROL, 12, 32, 64 },
+    /* CR13       */ { ZYDIS_REGCLASS_CONTROL, 13, 32, 64 },
+    /* CR14       */ { ZYDIS_REGCLASS_CONTROL, 14, 32, 64 },
+    /* CR15       */ { ZYDIS_REGCLASS_CONTROL, 15, 32, 64 },
+    /* DR0        */ { ZYDIS_REGCLASS_DEBUG, 0, 32, 64 },
+    /* DR1        */ { ZYDIS_REGCLASS_DEBUG, 1, 32, 64 },
+    /* DR2        */ { ZYDIS_REGCLASS_DEBUG, 2, 32, 64 },
+    /* DR3        */ { ZYDIS_REGCLASS_DEBUG, 3, 32, 64 },
+    /* DR4        */ { ZYDIS_REGCLASS_DEBUG, 4, 32, 64 },
+    /* DR5        */ { ZYDIS_REGCLASS_DEBUG, 5, 32, 64 },
+    /* DR6        */ { ZYDIS_REGCLASS_DEBUG, 6, 32, 64 },
+    /* DR7        */ { ZYDIS_REGCLASS_DEBUG, 7, 32, 64 },
+    /* DR8        */ { ZYDIS_REGCLASS_DEBUG, 8, 32, 64 },
+    /* DR9        */ { ZYDIS_REGCLASS_DEBUG, 9, 32, 64 },
+    /* DR10       */ { ZYDIS_REGCLASS_DEBUG, 10, 32, 64 },
+    /* DR11       */ { ZYDIS_REGCLASS_DEBUG, 11, 32, 64 },
+    /* DR12       */ { ZYDIS_REGCLASS_DEBUG, 12, 32, 64 },
+    /* DR13       */ { ZYDIS_REGCLASS_DEBUG, 13, 32, 64 },
+    /* DR14       */ { ZYDIS_REGCLASS_DEBUG, 14, 32, 64 },
+    /* DR15       */ { ZYDIS_REGCLASS_DEBUG, 15, 32, 64 },
+    /* K0         */ { ZYDIS_REGCLASS_MASK, 0, 64, 64 },
+    /* K1         */ { ZYDIS_REGCLASS_MASK, 1, 64, 64 },
+    /* K2         */ { ZYDIS_REGCLASS_MASK, 2, 64, 64 },
+    /* K3         */ { ZYDIS_REGCLASS_MASK, 3, 64, 64 },
+    /* K4         */ { ZYDIS_REGCLASS_MASK, 4, 64, 64 },
+    /* K5         */ { ZYDIS_REGCLASS_MASK, 5, 64, 64 },
+    /* K6         */ { ZYDIS_REGCLASS_MASK, 6, 64, 64 },
+    /* K7         */ { ZYDIS_REGCLASS_MASK, 7, 64, 64 },
+    /* BND0       */ { ZYDIS_REGCLASS_BOUND, 0, 128, 128 },
+    /* BND1       */ { ZYDIS_REGCLASS_BOUND, 1, 128, 128 },
+    /* BND2       */ { ZYDIS_REGCLASS_BOUND, 2, 128, 128 },
+    /* BND3       */ { ZYDIS_REGCLASS_BOUND, 3, 128, 128 },
+    /* BNDCFG     */ { ZYDIS_REGCLASS_BOUND, -1, 0, 0 },
+    /* BNDSTATUS  */ { ZYDIS_REGCLASS_BOUND, -1, 0, 0 },
```

Comment:
This is a change from previous version. `BNDCFG` and `BNDSTATUS` used to report `ZYDIS_REGCLASS_INVALID`, so new version causes fuzzers to crash. It's also inconsistent with `ZydisRegisterClassLookupItem` which reports correct `BND0-BND3` range.

I will compare this implementation against previous one to find other possible changes.

Reply:
Here's a summary of all the breaking changes:
| What | Old | New | Opinion |
| --- | --- | --- | --- |
| RAX-R15 width16/32 | 0 | 64 | :x: |
| X87CONTROL, X87STATUS, X87TAG | ZYDIS_REGCLASS_INVALID | ZYDIS_REGCLASS_X87 | :x: |
| GDTR, LDTR, IDTR, TR | ZYDIS_REGCLASS_INVALID | ZYDIS_REGCLASS_TABLE | :heavy_check_mark: |
| K0-K7 width16/32/64 | 0 | 64 | :question: |
| BNDCFG, BNDSTATUS | ZYDIS_REGCLASS_INVALID | ZYDIS_REGCLASS_BOUND | :x: |

:x: - bringing old behavior back is desirable
:heavy_check_mark: - new behavior is desirable
:question: - change is good but mask registers are 16-bits on KNC and they share same exact names with AVX-512 64-bits registers. If we want to keep this change then TODO: Stop treating `ZYDIS_REGCLASS_MASK` as special case inside `ZydisIsRegisterOperandCompatible` (correct this check: `if (reg_width == 0 && (reg_class != ZYDIS_REGCLASS_MASK))`).

Reply:
Thanks for checking. Makes sense! I have to adjust my simple python script to reflect the suggested changes.

Reply:
One condition was missing in the new Python script which I had in the old Delphi based one.

Reply:
> but mask registers are 16-bits on KNC

Right. Forgot about this. This was the reason for the `0`. I think `64` still makes more sense as KNC essentially is dead.

Reply:
> Right. Forgot about this. This was the reason for the `0`. I think `64` still makes more sense as KNC essentially is dead.

I agree, it's better to prioritize most common case rather than return 0.

Reply:
I just pushed some changes. Just removed the second part of the check inside `ZydisIsRegisterOperandCompatible` and left the `if (reg_width == 0)` like it was.

Reply:
LGTM :)